### PR TITLE
expression: implement vectorized evaluation for `builtinDayOfYearSig`

### PR DESCRIPTION
**Pull request by 710698117**
### What problem does this PR solve? 
implement vectorized evaluation for builtinDayOfYearSig , for #12103

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMathFunc -run=BenchmarkVectorizedBuiltinTimeFunc
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinTimeFunc/builtinDayOfYearSig-VecBuiltinFunc-4                               	  100000	     22484 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinDayOfYearSig-NonVecBuiltinFunc-4                            	   30000	     60245 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -1029,11 +1029,37 @@ func (b *builtinSubDateAndDurationSig) vecEvalString(input *chunk.Chunk, result
...
 func (b *builtinDayOfYearSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
```
Comment (XuHuaiyu) [inline expression/builtin_time_vec.go]: Should we also check `handleInvalidTimeError` here?

Comment (710698117) [reply expression/builtin_time_vec.go]: Yes, I've changed it.

**Comment by XuHuaiyu** [comment]
Hi, @710698117 
https://github.com/pingcap/tidb/pull/13091 has already implemented the vectorized `builtinDayOfYearSig` and been merged into the master branch.

Maybe you can choose another function to implement it?

**Comment by 710698117** [comment]
> Hi, @710698117
> #13091 has already implemented the vectorized `builtinDayOfYearSig` and been merged into the master branch.
> 
> Maybe you can choose another function to implement it?

Ok, thanks for informing.

**Pull request closed without merging**